### ISSUE TITLE
Scatter plot does not update

### ISSUE THREAD

**Issue by Nova-2119** [constellation-app/constellation#794]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help` > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/wiki/FAQ
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

The scatterplot view plot does not update in a large proportion of instances. Example when changing one of the axis. The user must close the view and then re-open it for the change to take effect.

### Steps to Reproduce

1. Open new Graph
2. Create Sphere Graph
3. Open ScatterPlot View
4. Change the X axis variable to X
5. Change the Y axis variable to Y
---- The below proves that changes also don't change the graph once a graph is visible
[Optional 1]  Close ScatterPlot View
[Optional 2] Open ScatterPlot View
[Optional 3] Change X or Y axis variable

**Expected behaviour:** [What you expect to happen]
The scatterplot should generate when the Y variable is selected. After it has generated the scatterplot should regenerate when options are changed

**Actual behaviour:** [What actually happens]
User must close and open view to generate initial plot and before seeing any changes to the plot.

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

**Comment by Nova-2119** [comment]
The scatter plot can be made to work by performing the following steps.
1. Open new graph
2. Create spher graph
3.Open scatterplot view
4. Set axis to X and Y attrbutes
5. Open map view
6. Close scatterplot view
7. Open scatterplot view

After doing the above the scatterplot should update in real time.

**Comment by Nova-2119** [comment]
Okay so my description of the problem wasn't quite accurate/detailed enough and I thin I have a fix.

The problem looks like it is specific to when the scatterPlot is opened after a graph has been opened. If you open a graph after the scatterPlot has opened it will act normally.
I believe I have fixed this by removing line 234 in https://github.com/constellation-app/constellation/blob/master/CoreScatterPlotView/src/au/gov/asd/tac/constellation/views/scatterplot/ScatterPlotTopComponent.java. When you open the scatterplotview with a graph already open this line runs and updates the currentGraph recorded by the scatterPlotTopComponent prior to it running newActiveGraph, which then says "oh hey this is the same graph no need to update the listener" meaning you have no listener for the scatterPlotTopComponent and hence it doesn't update.
I did one test and this seems to have fixed it, but I need to do some more testing next week to ensure it hasn't had an unintended knock on affect. Im not entirely sure when the method componentShowing (which has the deleted line in) is called so until I understand that I can't be confident in the potential knock on effects.
